### ISSUE TITLE
react mode indentation fails to handle chaining

### ISSUE THREAD

**Issue by tko** [syl20bnr/spacemacs#4920]
Mostly react mode indentation works just right, but I just realized it's failing when chaining calls, in two rather opposite ways.

On one hand using the vim `=` command does nothing to change the indentation of the first indented line in this case. Only the followup lines are reindented to match the first one.
```javascript
fetch(someVariable)
    .then()
  .then()
  .catch();
```

On the other hand if there's a dot on the first line, even if it's inside a comment, then the second line is aligned with the dot and the rest of lines follow. 
```javascript
fetch('http://example.com/')
                     .then()
                     .then()
                     .catch();
fetch(some.object)
          .then()
          .then()
          .catch();
fetch(someVariable) // fixme.
                            .then()
                            .then()
                            .catch();
```

Switching to `js2-mode` and reindenting the buffer indents all the lines with leadings dots indented by indentation offset as one would expect.

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.9
- Spacemacs branch: master (rev. 853d3d6)
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp
 (auto-completion :variables auto-completion-enable-help-tooltip t)
 syntax-checking version-control git markdown org
 (osx :variables osx-use-option-as-meta nil)
 (colors :variables colors-enable-rainbow-identifiers nil)
 python
 (haskell :variables haskell-enable-ghc-mod-support t haskell-process-type 'stack-ghci)
 html javascript react
 (c-c++ :variables c-c++-enable-clang-support t)
 restclient
 (puppet :variables puppet-use-smie t)
 yaml dash semantic go)
```

**Comment by StreakyCobra** [comment]
Could someone try to reproduce this on a raw Emacs installation, to know if it is an upstream bug or a Spacemacs one?

**Comment by TheBB** [comment]
Can't reproduce in the current develop.

```react
fetch(someVariable)
    .then()
    .then()
    .catch();

fetch('http://some.example.com')
    .then()
    .then()
    .catch();

fetch(some.object)
    .then()
    .then()
    .catch();

fetch(someVariable) // fix.me
    .then()
    .then()
    .catch();
```

Since nobody else appears to have this problem, I am tentatively marking this as fixed in develop (since there hasn't been a major release since the report was made).

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
